feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/autodiscovery/view/AutoDiscoveryResultView.kt

```diff
@@ -36,7 +36,7 @@ internal fun AutoDiscoveryResultView(
                     color = Color.Gray.copy(alpha = 0.5f),
                     shape = MainTheme.shapes.small,
                 )
-                .clickable { expanded.value = !expanded.value },
+                .clickable(enabled = settings != null) { expanded.value = !expanded.value },
```

Comment:
This leads to the correct result, but it's not immediately obvious why that is. 

I think it'd be cleaner if we moved the code to select `AutoDiscoveryResultHeaderState` further up, assigned the result to a variable, and used the `isExpandable` property for the enable/disable decision.

```kotlin
val discoveryResultHeaderState = if (settings == null) {
    AutoDiscoveryResultHeaderState.NoSettings
} else if (settings.isTrusted) {
    AutoDiscoveryResultHeaderState.Trusted
} else {
    AutoDiscoveryResultHeaderState.Untrusted
}

Surface(
    shape = MainTheme.shapes.small,
    modifier = Modifier
        .border(
            width = 1.dp,
            color = Color.Gray.copy(alpha = 0.5f),
            shape = MainTheme.shapes.small,
        )
        .clickable(enabled = discoveryResultHeaderState.isExpandable) { expanded.value = !expanded.value },
) {
    // …
}
```

Reply:
@cketti Done!